Resolving host takes too long compared to other browsers or original chromium

**Describe the bug**
Accessing a website that has not been opened yet in the current session always takes 1-2 seconds before it starts loading making the browsing experince terrible. The status bar at the bottom of the browser says "Resolving host" during this wait, so my assumption is that there're something slowing DNS queries down. 

What I've tried:
- --set-ipv6-probe-false
- removing ~/.config/chromium dir for a fresh install
- other browsers work just fine
- even original Chromium works ok
- different networks
- changing DNS servers

**To Reproduce**
As I couldn't find any open issue that would describe my problem, I assume it is for some reason specific to my machine. I'm not aware of anything special on my computer however. I'm not even using a system-wide DNS cache. No IPv6.

**Environment (please complete the following information):**
 - OS/Platform and version: Linux version 5.8.5-arch1-1, Arch Linux, i3
 - ungoogled-chromium version: Version 84.0.4147.125 (Official Build) Arch Linux (64-bit)

--
**Could anyone push me in the right direction so that I can figure out what's the problem, please?**

@Ahrotahn thanks for the detailed reply.

**Build failure:**

Well, after 6+ hours of building from source, there was a power failure for a few minutes. So I tried to restart the build with:
````
dpkg-buildpackage -b -uc -nc
````
Though ultimately it failed with alot of errors, here are the last few messages:
````
ld.lld: error: too many errors emitted, stopping now (use -error-limit=0 to see all errors)
clang: error: linker command failed with exit code 1 (use -v to see invocation)
ninja: build stopped: subcommand failed.
make[1]: *** [debian/rules:143: override_dh_auto_build-arch] Error 1
make[1]: Leaving directory '/home/admn/Downloads/build/src'
make: *** [debian/rules:119: binary] Error 2
dpkg-buildpackage: error: debian/rules binary subprocess returned exit status 2
````
Here's a pastebin with all the errors i got:
`https://pastebin.com/iNiFk1UX`

**Warnings:**

Even before the power failures after 6+ hours, I got these warnings, which may or may not have anything to do with build failure.
````
ninja -j12 -C out/Release chrome chrome_sandbox content_shell chromedriver

ninja: Entering directory `out/Release'

[16967/39673] CXX obj/media/gpu/vaapi/common/vaapi_wrapper.o

../../media/gpu/vaapi/vaapi_wrapper.cc:437:5: warning: misleading indentation; statement is not part of the previous 'if' [-Wmisleading-indentation]
    const char kNvidiaPath[] = "/dev/dri/nvidiactl";
    ^

../../media/gpu/vaapi/vaapi_wrapper.cc:435:3: note: previous statement is here
  if (drm_file.IsValid())
  ^
1 warning generated.

[26663/39673] CXX obj/chrome/browser/browser/browser_signin_policy_handler.o

../../chrome/browser/policy/browser_signin_policy_handler.cc:52:13: warning: enumeration values 'kDisabled' and 'kEnabled' not handled in switch [-Wswitch]
    switch (static_cast<BrowserSigninMode>(int_value)) {
            ^
1 warning generated.
````

**Removing the patch:**

(1) Did the following 2, right before starting the building (`dpkg-buildpackage -b -uc`)
(2) Removed 2 `add-ipv6-probing-option.patch` files from `build` directory, and 
(3) Deleted the `line 63` from these two files:
````
/build/src/debian/patches/series

build/src/debian/ungoogled-upstream/ungoogled-chromium/patches/series
````
**Internal host resolver:**

> Chromium (and most major browsers these days) has it's own internal host resolver.

> Zoraver had trouble with disabling mDNS on the last major version update and it would be just as much trouble trying to disable the inbuilt DNS for the same reasons. All of the host resolver stuff is a moving target right now since Google is attempting to make DoH default, so most solutions would likely be short-lived.

This is the most troubling part to me. For me personally, few seconds in DNS lookup is not a big deal but google hardcoding its own DNS servers is a big deal. Well, I would be back to Firefox, 'cause Firefox uses system / Unbound DNS resolver.

Though Firefox does support DNS over HTTPS, but I can disable it completely with `network.trr.mode=5` in `about:config`

Thank you all.

--- a/docs/flags.md
+++ b/docs/flags.md
@@ -21,5 +21,10 @@ If a flag requires a value, you must specify it with an `=` sign; e.g. flag `--f
 * `--omnibox-autocomplete-filtering` - Restrict omnibox autocomplete results to search suggestions (if enabled) or search suggestions and bookmarks. The type of filtering is determined by the values `search-suggestions-only` and `search-suggestions-and-bookmarks`, respectively.
 * `--pdf-plugin-name` - Sets the internal PDF viewer plugin name. Useful for sites that probe JavaScript API `navigator.plugins`. Supports values `chrome` for Chrome, `edge` for Microsoft Edge. Default value when omitted is Chromium.
 * `--scroll-tabs` - Determines if scrolling will cause a switch to a neighboring tab if the cursor hovers over the tabs, or the empty space beside the tabs. The flag requires one the values: `always`, `never`, `incognito-and-guest`. When omitted, the default is to use platform-specific behavior, which is currently enabled only on desktop Linux.
-* `--set-ipv6-probe-false` - (Not in `chrome://flags`) Forces the result of the browser's IPv6 probing (i.e. IPv6 connectivity test) to be unsuccessful. This causes IPv4 addresses to be prioritized over IPv6 addresses. Without this flag, the probing result is set to be successful, which causes IPv6 to be used over IPv4 when possible.
 * `--show-avatar-button` - Sets visibility of the avatar button. The flag requires one of the values: `always`, `incognito-and-guest` (only show Incognito or Guest modes), or `never`.
+
+## Feature flags
+
+Feature flags are enabled with the `--enable-features` switch.  Multiple features can be passed at the same time by separating them with a comma, e.g. `--enable-features=flag1,flag2,flag3`.
+
+* `SetIpv6ProbeFalse` - (Not in `chrome://flags`) Forces the result of the browser's IPv6 probing (i.e. IPv6 connectivity test) to be unsuccessful. This causes IPv4 addresses to be prioritized over IPv6 addresses. Without this flag, the probing result is set to be successful, which causes IPv6 to be used over IPv4 when possible.

--- a/patches/extra/ungoogled-chromium/add-ipv6-probing-option.patch
+++ b/patches/extra/ungoogled-chromium/add-ipv6-probing-option.patch
@@ -1,17 +1,30 @@
 # Disables IPv6 probing and adds an option to change the IPv6 probing result
 # TODO: Consider adding a chrome://flag to set the command-line flag
 
+--- a/net/base/features.cc
++++ b/net/base/features.cc
+@@ -164,5 +164,7 @@
+ const base::Feature kUseLookalikesForNavigationSuggestions{
+     "UseLookalikesForNavigationSuggestions", base::FEATURE_DISABLED_BY_DEFAULT};
+ 
++const base::Feature kSetIpv6ProbeFalse{"SetIpv6ProbeFalse", base::FEATURE_DISABLED_BY_DEFAULT};
++
+ }  // namespace features
+ }  // namespace net
+--- a/net/base/features.h
++++ b/net/base/features.h
+@@ -245,6 +245,8 @@
+ // locally-generated suggestions to visit similar domains.
+ NET_EXPORT extern const base::Feature kUseLookalikesForNavigationSuggestions;
+ 
++NET_EXPORT extern const base::Feature kSetIpv6ProbeFalse;
++
+ }  // namespace features
+ }  // namespace net
+ 
 --- a/net/dns/host_resolver_manager.cc
 +++ b/net/dns/host_resolver_manager.cc
-@@ -30,6 +30,7 @@
- #include "base/bind_helpers.h"
- #include "base/callback.h"
- #include "base/callback_helpers.h"
-+#include "base/command_line.h"
- #include "base/compiler_specific.h"
- #include "base/containers/flat_set.h"
- #include "base/containers/linked_list.h"
-@@ -131,11 +132,6 @@ const unsigned kMinimumTTLSeconds = kCac
+@@ -131,11 +131,6 @@ const unsigned kMinimumTTLSeconds = kCac
  // cached.
  const int kIPv6ProbePeriodMs = 1000;
  
@@ -23,12 +36,12 @@
  enum DnsResolveStatus {
    RESOLVE_STATUS_DNS_SUCCESS = 0,
    RESOLVE_STATUS_PROC_SUCCESS,
-@@ -3584,7 +3580,7 @@ bool HostResolverManager::IsIPv6Reachabl
+@@ -3584,7 +3579,7 @@ bool HostResolverManager::IsIPv6Reachabl
        (tick_clock_->NowTicks() - last_ipv6_probe_time_).InMilliseconds() >
            kIPv6ProbePeriodMs) {
      SetLastIPv6ProbeResult(
 -        IsGloballyReachable(IPAddress(kIPv6ProbeAddress), net_log));
-+        !base::CommandLine::ForCurrentProcess()->HasSwitch("set-ipv6-probe-false"));
++        !base::FeatureList::IsEnabled(features::kSetIpv6ProbeFalse));
      cached = false;
    }
    net_log.AddEvent(

